src/Processors/QueryPlan/Optimizations/optimizeJoinByShards.cpp

```diff
@@ -126,8 +126,25 @@ JoinStep::PrimaryKeySharding findCommonPrimaryKeyPrefixByJoinKey(
 
     JoinStep::PrimaryKeySharding sharding;
 
+    bool first = true;
     for (size_t pos = 0; pos < lhs_pk_colum_names.size() && pos < rhs_pk_colum_names.size(); ++pos)
     {
+        bool ldesc = (pos < lhs_pk.reverse_flags.size()) ? lhs_pk.reverse_flags[pos] : false;
+        bool rdesc = (pos < rhs_pk.reverse_flags.size()) ? rhs_pk.reverse_flags[pos] : false;
+        if (ldesc != rdesc)
```

Comment:
Technically, it's possible to support PKs with a different sorting order.
For now, this small fix is better.